src/flint/types/fq_default_poly.pyx

```diff
@@ -0,0 +1,1643 @@
+
+    def reverse(self, degree=None):
+        """
+        Return a polynomial with the coefficients of this polynomial
+        reversed.
+
+        If ``degree`` is not None, the output polynomial will be zero-padded
+        or truncated before being reversed. NOTE: degree must be non-negative.
+
+            >>> R = fq_default_poly_ctx(163, 3)
+            >>> f = R([1,2,3,4,5])
+            >>> f.reverse()
+            x^4 + 2*x^3 + 3*x^2 + 4*x + 5
+            >>> f.reverse(degree=1)
+            x + 2
+            >>> f.reverse(degree=100)
+            x^100 + 2*x^99 + 3*x^98 + 4*x^97 + 5*x^96
+        """
+        cdef fq_default_poly res
+        cdef slong d
+
+        if degree is not None:
+            d = degree
+            if d != degree or d < 0:
+                raise ValueError(f"degree argument must be a non-negative integer, got {degree}")
+        else:
+            d = fq_default_poly_degree(self.val, self.ctx.field.val)
+
+        length = d + 1
+
+        res = self.ctx.new_ctype_poly()
+        fq_default_poly_reverse(res.val, self.val, length, self.ctx.field.val)
+        return res
+
+    def truncate(self, slong n):
+        """
+        Notionally truncate the polynomial to have length ``n``. If
+        ``n`` is larger than the length of the input, then ``self`` is
+        returned. If ``n`` is not positive, then the zero polynomial
+        is returned.
+
+        Effectively returns this polynomial :math:`\mod x^n`.
+
+            >>> R = fq_default_poly_ctx(163, 5)
+            >>> f = R([1,2,3])
+            >>> f.truncate(3) == f
+            True
+            >>> f.truncate(2)
+            2*x + 1
+            >>> f.truncate(1)
+            1
+            >>> f.truncate(0)
+            0
+            >>> f.truncate(-1)
+            0
+        """
+        cdef fq_default_poly res
+
+        length = fq_default_poly_degree(self.val, self.ctx.field.val)
+        res = self.ctx.new_ctype_poly()
+
+        if n <= 0: # return zero
+            return res
+        elif n > length: # do nothing
+            fq_default_poly_set(
+                res.val, self.val, self.ctx.field.val
+            )
+        else:
+            fq_default_poly_set_trunc(
+                res.val, self.val, n, self.ctx.field.val
+            )
+        return res
+
+    def monic(self, check=True):
+        """
+        Return this polynomial divided by its leading coefficient.
+
+        If ``check`` is True, raises ValueError if the leading coefficient
+        is not invertible modulo N. If ``check`` is False and the leading
+        coefficient is not invertible, the output is undefined.
+
+            >>> R = fq_default_poly_ctx(163)
+            >>> f = R([1,2,3])
+            >>> f.monic()
+            x^2 + 55*x + 109
+        """
+        cdef fq_default_poly res
+
+        res =  self.ctx.new_ctype_poly()
+        fq_default_poly_make_monic(
+            res.val, self.val, self.ctx.field.val
+        )
+        return res
```

Comment:
It doesn't look like `check` is used.

Reply:
Oh this is just a copy paste from fmpz_mod_poly which I did not clean up. Thanks for spotting it.